### PR TITLE
lib: at_host: Socket thread change

### PR DESCRIPTION
**Pull request by martiles**
This change is needed to make the at_host library
handle the at sockets correctly.

Signed-off-by: Martin Lesund <martin.lesund@nordicsemi.no>

### CODE REVIEW

**Comment by martiles** [comment]
@jtguggedal

File: lib/at_host/at_host.c [hunk 1]
```diff
@@ -194,32 +194,21 @@ static int at_uart_init(char *uart_dev_name)
...
 
 	while (1) {
-		/* Poll the socket for incoming data. */
-		err = poll(fds, nfds, K_FOREVER);
-		if (err < 0) {
-			LOG_ERR("Poll error: %d\n", err);
-		}
-
-		k_mutex_lock(&socket_mutex, K_FOREVER);
 		/* Read AT socket in non-blocking mode. */
```
Comment (jtguggedal) [inline lib/at_host/at_host.c]: This comment should now say that it's reading in blocking mode

File: lib/at_host/at_host.c [hunk 2]
```diff
@@ -194,32 +194,21 @@ static int at_uart_init(char *uart_dev_name)
...
-		k_mutex_unlock(&socket_mutex);
+				sizeof(at_read_buff), 0);
 
 		/* Forward the data over UART if any. */
-		/* If no data, errno is set to EGAIN and we will try again. */
 		if (r_bytes > 0) {
 			/* Poll out what is in the buffer gathered from
-			 * the modem.
-			 */
+			 * the modem.*/
```
Comment (jtguggedal) [inline lib/at_host/at_host.c]: A space can be added before the closing asterisk

File: lib/at_host/at_host.c [hunk 3]
```diff
@@ -194,28 +194,18 @@ static int at_uart_init(char *uart_dev_name)
...
 	ARG_UNUSED(arg3);
 
 	while (1) {
-		/* Poll the socket for incoming data. */
-		err = poll(fds, nfds, K_FOREVER);
-		if (err < 0) {
-			LOG_ERR("Poll error: %d\n", err);
-		}
-
-		k_mutex_lock(&socket_mutex, K_FOREVER);
```
Comment (rlubos) [inline lib/at_host/at_host.c]: We should get rid of `socket_mutex` entirely then - there's no need to use it for tx-only.

Comment (martiles) [reply lib/at_host/at_host.c]: The mutexes should now be removed

**Comment by martiles** [comment]
The issue with the freezing appeared after the commit: 4753d81a2c3f9033449295847ebc396c91412189
However, the 'specific' reason why this affected the behaviour of this library, I do not know.

**Pull request merged**